cscs-checks/prgenv/helloworld.py

```diff
@@ -11,34 +11,43 @@
 
 
 class HelloWorldBaseTest(rfm.RegressionTest):
-    def __init__(self, variant, lang, linkage):
-        self.linkage = linkage
-        self.variables = {'CRAYPE_LINK_TYPE': linkage}
-        self.prgenv_flags = {}
-        self.lang_names = {
-            'c': 'C',
-            'cpp': 'C++',
-            'f90': 'Fortran 90'
-        }
-        self.descr = f'{self.lang_names[lang]} Hello World'
-        self.sourcepath = 'hello_world'
-        self.build_system = 'SingleSource'
-        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
-                              'arolla:cn', 'arolla:pn', 'tsa:cn', 'tsa:pn']
-        if linkage == 'dynamic':
+    linking = parameter(['dynamic', 'static'])
+    lang = parameter(['c', 'cpp', 'f90'])
+    prgenv_flags = {}
+    lang_names = {
+        'c': 'C',
+        'cpp': 'C++',
+        'f90': 'Fortran 90'
+    }
+    sourcepath = 'hello_world'
+    build_system = 'SingleSource'
+    valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                     'arolla:cn', 'arolla:pn', 'tsa:cn', 'tsa:pn']
+    valid_prog_environs = ['PrgEnv-aocc', 'PrgEnv-cray',
+                           'PrgEnv-cray_classic', 'PrgEnv-gnu',
+                           'PrgEnv-intel', 'PrgEnv-pgi',
+                           'PrgEnv-gnu-nocuda', 'PrgEnv-pgi-nocuda']
+    exclusive_access = True
+    compilation_time_seconds = None
```

Comment:
Shouldn't that be a variable and not a class variable? Each test is setting that.